### PR TITLE
Auto tagging of partition key fields

### PR DESCRIPTION
**Pull request by saroh**
Refs #3000

### Description

Partition key fields are added to tag fields.

:warning: Fails retrocompatibility tests where the partition key is not defined in the field mappings, hence cannot be validated as a tag. Cf third commit. Is this ok or not? 

### How was this PR tested?

UT.

### CODE REVIEW

File: quickwit/quickwit-metastore/test-data/file-backed-index/v0.4-1cfee07237d8cbbe9240d5a883bf98c2.expected.json [hunk 1]
```diff
@@ -62,7 +62,7 @@
         ],
         "max_num_partitions": 100,
         "mode": "dynamic",
-        "partition_key": "tenant",
```
Comment (saroh) [inline quickwit/quickwit-metastore/test-data/file-backed-index/v0.4-1cfee07237d8cbbe9240d5a883bf98c2.expected.json:65]: There is no field declared as `tenant`

**Comment by fulmicoton** [comment]
Awesome! Thank you @saroh. Give me a bit of time to review this!

File: quickwit/quickwit-config/src/index_config/mod.rs [hunk 2]
```diff
@@ -427,7 +427,7 @@ impl TestableForRegression for IndexConfig {
             store_source: true,
             mode: ModeType::Dynamic,
             dynamic_mapping: None,
-            partition_key: Some("tenant".to_string()),
```
Comment (saroh) [inline quickwit/quickwit-config/src/index_config/mod.rs:430]: Field not declared in mapping.

**Comment by saroh** [comment]
> Awesome! Thank you @saroh. Give me a bit of time to review this!

Np, I'm unsure of the retro compatibility changes. I can update the doc accordingly once I know if this change is OK or breaking :D

**Comment by fulmicoton** [comment]
@saroh no need to worry about generated file. We can deal with those.

**Comment by fulmicoton** [comment]
I think you have the permission to create branches on quickwit-oss. Please do so, it makes it easier to checkout, add extra commits etc.

**Comment by fulmicoton** [comment]
Closing for #3449

**Pull request closed without merging**